### PR TITLE
#170932537 Challenge 3 setup + code refactoring

### PR DESCRIPTION
**Pull request by the22mastermind**
#### What does this PR do?
Code refactoring and challenge 3 setup
#### Description of Task to be completed?
1. Code base refactoring
2. Install PostgreSQL
3. Configure database connection
4. Create database and tables
5. Create scripts for creating and dropping tables
#### What are the relevant pivotal tracker stories?
#170932537
#### Questions:

### CODE REVIEW

File: server/config/db.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+/* eslint-disable import/no-mutable-exports */
+import pg from 'pg';
+import dotenv from 'dotenv';
+
+dotenv.config();
```
Comment (alainburindi) [inline server/config/db.js:5]: setting up dotenv is done only once. basically in the entry point of your project. then you no longer need to set it up

File: server/config/tables.js [hunk 2]
```diff
@@ -0,0 +1,93 @@
+/* eslint-disable no-console */
+import pool from './db';
+
+pool.on('connect', () => {
+  console.log('Connected to PSQL DB');
+});
```
Comment (alainburindi) [inline server/config/tables.js]: Would you explain why you put this instruction??

File: server/config/tables.js [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+// Drop Tables
+const drop = () => {
+  const users = 'DROP TABLE IF EXISTS users CASCADE;';
+  const announcements = 'DROP TABLE IF EXISTS announcements CASCADE;';
+  const flags = 'DROP TABLE IF EXISTS flags CASCADE;';
+  const con = `${users}; ${announcements}; ${flags};`;
```
Comment (alainburindi) [inline server/config/tables.js]: Would you mind renaming these variables properly?? a name that represents what they store

File: server/config/tables.js [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+    })
+    .catch((err) => {
+      console.log(err);
+      pool.end();
+    });
+
+  pool.on('remove', () => {
+    console.log('client removed');
+    process.exit(0);
+  });
```
Comment (alainburindi) [inline server/config/tables.js]: Remove these `console.log()`

File: server/config/tables.js [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
```
Comment (alainburindi) [inline server/config/tables.js]: consider the previous comment about the logs

**Review by alainburindi — CHANGES_REQUESTED** [review]

File: server/config/tables.js (hunk 2 above)
Comment (the22mastermind) [reply server/config/tables.js]: To track the progress of the script when I run it. I want to see it if connecting to db is successful, response from creating or dropping tables, then closing the connection.

File: server/config/tables.js (hunk 3 above)
Comment (the22mastermind) [reply server/config/tables.js]: Noted.

File: server/config/tables.js (hunk 4 above)
Comment (the22mastermind) [reply server/config/tables.js]: Noted.

**Pull request merged**